src/lib/rest/RestService.cpp

```diff
@@ -52,9 +52,111 @@
 #define MAX_TENANT_NAME_LEN_STRING    "50"
 
 
+static std::string textParseAttributeValue(ConnectionInfo* ciP, ContextAttribute* caP)
+{
+  double d;
+
+  // 1. Starts and ends with citation marks?
+  if (ciP->payload[0] == '"')
+  {
+    char* end = &ciP->payload[strlen(ciP->payload) - 1];
+
+    if (*end == '"')
+    {
+      *end = 0;
+      caP->stringValue = &ciP->payload[1];
+      caP->valueType   = orion::ValueTypeString;
+    }
+    else
+    {
+      OrionError oe(SccBadRequest, "Missing citation-mark at end of string");
+
+      return oe.render(ciP, "");
+    }
+  }
+
+  // 2. True or false?
+  else if ((strlen(ciP->payload) == 4) && ((strcmp(ciP->payload, "true") == 0) || (strcmp(ciP->payload, "True") == 0) || (strcmp(ciP->payload, "TRUE") == 0)))
+  {
+    caP->boolValue   = true;
+    caP->valueType   = orion::ValueTypeBoolean;
+  }
+  else if ((strlen(ciP->payload) == 5) && ((strcmp(ciP->payload, "false") == 0) || (strcmp(ciP->payload, "False") == 0) || (strcmp(ciP->payload, "FALSE") == 0)))
+  {
+    caP->boolValue   = false;
+    caP->valueType   = orion::ValueTypeBoolean;
+  }
+
+  // 3. Null ?
+  else if ((strlen(ciP->payload) == 4) && ((strcmp(ciP->payload, "null") == 0) || (strcmp(ciP->payload, "Null") == 0) || (strcmp(ciP->payload, "NULL") == 0)))
+  {
+    caP->valueType   = orion::ValueTypeNull;
+  }
+
+  //
+  // 4. Valid Double?
+  // FIXME P4: this is much more complex than just (atof(string) != 0 || string == "0")
+  //           0.000 also is a valid float and it given 0 - 0.0000000000, and 0e0 also ...
+  //           And, even worse, 123K gives 123.0 back, we would need to analyze the string to try to
+  //           find garbage bytes after it if we want to detect this error.
+  //           However, all of this is not so extremely important and for now, (atof(string) != 0 || string== "0") is good enough
+  //
+  else if (((d = strtod(ciP->payload, NULL)) != 0) || ((ciP->payload[0] == '0') || (ciP->payload[1] == 0)))
```

Comment:
I'm afraid that I don't have the solution... on the contrary (if I'm remembering correctly) I have the same "0 gap" problem in other parts of the code :( 

My only advice here is to study how other C/C++ projects have solved the same problem. It no solution comes in a reasonable time, them create an issue about this (not only for textParseAttributeValue(), but also to analyze in which other points of the code we have the same problem) and live with that FIXME while we find a solution.